.github/workflows/build.yml

```diff
@@ -117,4 +118,5 @@ jobs:
           tags: |
             docker.io/hashicorp/${{env.repo}}:${{env.version}}
           dev_tags: |
-            docker.io/hashicorppreview/${{env.repo}}:${{env.version}}
+            docker.io/hashicorppreview/${{env.repo}}:${{env.version}}-dev
+            docker.io/hashicorppreview/${{env.repo}}:${{env.version}}-dev-${{github.sha}}
```

Comment:
Could we slice off the patch segment so this is just `MAJOR.MINOR-dev` to reflect the intended change in https://github.com/hashicorp/consul/pull/13308?

Reply:
I love this idea; however, my focus this PR is just getting the `main` branch published to unblock other work.

Mind if we circle back with a targeted PR to change how we're tagging for `release/*` branches?